app/src/main/java/com/duckduckgo/app/generalsettings/showonapplaunch/ShowOnAppLaunchViewModel.kt

```diff
@@ -19,57 +19,62 @@ package com.duckduckgo.app.generalsettings.showonapplaunch
 import com.duckduckgo.common.utils.DispatcherProvider
 import com.duckduckgo.di.scopes.ActivityScope
 import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.filterNotNull
-import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
 @ContributesViewModel(ActivityScope::class)
 class ShowOnAppLaunchViewModel @Inject constructor(
-    dispatcherProvider: DispatcherProvider,
+    @AppCoroutineScope private val appScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
+    private val showOnAppLaunchOptionDataStore: ShowOnAppLaunchOptionDataStore,
 ) : ViewModel() {
 
-    sealed class ShowOnAppLaunchOption {
-
-        data object LastOpenedTab : ShowOnAppLaunchOption()
-        data object NewTabPage : ShowOnAppLaunchOption()
-        data class SpecificPage(val url: String) : ShowOnAppLaunchOption()
-    }
-
     data class ViewState(
         val selectedOption: ShowOnAppLaunchOption,
+        val specificPageUrl: String,
     )
 
     private val _viewState = MutableStateFlow<ViewState?>(null)
     val viewState = _viewState.asStateFlow().filterNotNull()
 
     init {
-        viewModelScope.launch(dispatcherProvider.io()) {
-            // TODO get selected option from prefs
+        observeShowOnAppLaunchOptionChanges()
+    }
 
-            _viewState.value = ViewState(
-                selectedOption = LastOpenedTab,
-            )
-        }
+    private fun observeShowOnAppLaunchOptionChanges() {
+        combine(
+            showOnAppLaunchOptionDataStore.optionFlow,
+            showOnAppLaunchOptionDataStore.specificPageUrlFlow,
+        ) { option, specificPageUrl ->
+            _viewState.value = ViewState(option, specificPageUrl)
+        }.flowOn(dispatcherProvider.io())
+            .launchIn(viewModelScope)
     }
 
     fun onShowOnAppLaunchOptionChanged(option: ShowOnAppLaunchOption) {
         Timber.i("User changed show on app launch option to $option")
-        when (option) {
-            LastOpenedTab -> _viewState.update { it?.copy(selectedOption = option) }
-            NewTabPage -> _viewState.update { it?.copy(selectedOption = option) }
-            is SpecificPage -> {
-                // TODO get the last set page if we have one and populate the url
-                _viewState.update { it?.copy(selectedOption = SpecificPage("duckduckgo.com")) }
-            }
+        viewModelScope.launch {
+            showOnAppLaunchOptionDataStore.setShowOnAppLaunchOption(option)
+        }
+    }
+
+    fun setSpecificPageUrl(url: String) {
+        Timber.i("Setting specific page url to $url")
+        appScope.launch {
```

Comment:
Do you need the `appScope` here?

You can do:
```
viewModelScope.launch(dispatcherProvider.io()) {
            showOnAppLaunchOptionDataStore.setSpecificPageUrl(url)
        }
```

Along with these changes in `ShowOnAppLaunchActivity`:

`.flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)`  (STARTED instead of RESUMED)

and
```
    override fun onPause() {
        super.onPause()
        viewModel.setSpecificPageUrl(binding.specificPageUrlInput.text) 
    }
``` 
(instead of `override fun onStop()`)

Reply:
I'll take a look into this, I definitely managed to get into a state where the entered text would not be there after killing the process but maybe these suggestions help 🙏

Reply:
I can only reproduce the issue in one scenario, click recents, kill the process. I don't think is is worth the time or effort, as most of the time people will navigate away or go to the home screen. I'll remove appScope.

Reply:
Comments have been addressed and I made the changes you suggested, thank you 🙏 

I remember where I was seeing the issue now, it was not process death but pressing the up button, but moving to onPause seems to have solved it. Oh back to the fun of Lifecycles 🙃